src/Mvc/Mvc.Razor/src/RazorPageBase.cs

```diff
@@ -26,74 +25,26 @@ namespace Microsoft.AspNetCore.Mvc.Razor
     /// <summary>
     /// Represents properties and methods that are needed in order to render a view that uses Razor syntax.
     /// </summary>
-    public abstract class RazorPageBase : IRazorPage
+    public abstract class RazorPageBase : RazorRazorPageBase<ViewContext>, IRazorPage
```

Comment:
At a base level I really don't think we should try to make MVC inherit from this new thing. My concern is that if we're going to create a new thing, but have it inherit all of MVC's quirks and requirements. Basically, I think it's a success criteria to try and factor out the portable set of features and expose them through a reasonable design. I don't think it's a success criteria to have code-sharing. 

I'm going to try and focus my feedback on the design of the new thing, and once that's nailed it would be good to talk about what MVC does.